Classes/Systems/PhotoViewHandler.swift

```diff
@@ -8,24 +8,36 @@
 
 import UIKit
 import NYTPhotoViewer
+import IGListKit
+import SDWebImage
 
 final class PhotoViewHandler: NSObject,
     IssueCommentImageCellDelegate,
 NYTPhotosViewControllerDelegate {
 
     private weak var viewController: UIViewController? = nil
     private weak var referenceImageView: UIImageView? = nil
-
-    init(viewController: UIViewController?) {
+    private var imageModels: [IssueCommentImageModel]?
+    
+    init(viewController: UIViewController?, models: [ListDiffable]) {
         self.viewController = viewController
+        self.imageModels = models.flatMap({ $0 as? IssueCommentImageModel })
     }
 
     // MARK: IssueCommentImageCellDelegate
 
-    func didTapImage(cell: IssueCommentImageCell, image: UIImage) {
+    func didTapImage(cell: IssueCommentImageCell, image: UIImage, url: URL) {
         referenceImageView = cell.imageView
-        let photo = IssueCommentPhoto(image: image)
-        let photosViewController = NYTPhotosViewController(photos: [photo])
+        
+        var photos = [IssueCommentPhoto(image: image)]
+        
+        imageModels?.forEach({
+            guard $0.url.absoluteString != url.absoluteString else { return }
+            guard let image = SDImageCache.shared().imageFromCache(forKey: $0.url.absoluteString) else { return }
+            photos.append(IssueCommentPhoto(image: image))
+        })
```

Comment:
noting to myself that this implies that `SDImageCache` will always be used. Ok for now.

Reply:
Like I said was just toying around to see what was possible - completely understand if these limitations are too much

NYT is definitely a limiting factor here as they require an image straight up and don't allow any sort of async loading. Maybe a local fork or alternative library would be better suited - Card it up?